brax/training/agents/ppo/train.py

```diff
@@ -322,10 +328,13 @@ def f(carry, unused_t):
     assert data.discount.shape[1:] == (unroll_length,)
 
     # Update normalization params and normalize observations.
-    normalizer_params = running_statistics.update(
-        training_state.normalizer_params,
-        data.observation,
-        pmap_axis_name=_PMAP_AXIS_NAME)
+    if normalize_observations:
```

Comment:
why do we need this if statement, and the one that was added below it?

Reply:
For line 392, this handles the case when obs doesn't have the ['state'] key. Not sure if it makes sense for us to cover this case.

For the block under 331, there's nothing to update the normaliser with if 'state' isn't in the obs dict.

Reply:
I think these if statements should be removed if the purpose is to validate whether "state" exists or not.

Let's fail with KeyError, as discussed previously, we don't want to fail silently in general